random/src/range.rs

```diff
@@ -60,6 +66,36 @@ impl UniformU64Sampler {
     }
 }
 
+/// Sample a random number in `range` using `rng` as generator for random `u64` numbers
+///
+/// This utility exists only to provide compatibility of sampling algorithm with `rand`
+/// library at versions <=0.8.5, it is equivalent to its `rng.gen_range(range)`.
+///
+/// Panics: when `range` is empty.
+pub fn random_u64_range(rng: &mut impl Rng, range: impl RangeBounds<u64>) -> u64 {
+    let start = match range.start_bound() {
+        Bound::Unbounded => 0,
+        Bound::Included(start) => *start,
+        Bound::Excluded(&u64::MAX) => panic!("Cannot generate number in empty range (max..)"),
+        Bound::Excluded(start) => start.wrapping_add(1),
+    };
+    let last = match range.end_bound() {
+        Bound::Unbounded | Bound::Included(&u64::MAX) if start == 0 => return rng.gen(),
```

Comment:
Can we not set last to u64::MAX?

Reply:
The issue is that sampler API takes an exclusive range end, so it actually do not support getting a full (or right open) range.

This would trigger issue in the code below:
```
    // last - start != u64::MAX after check calculating last above, so +1 won't overflow
    let zero_range_end = NonZero::new(zero_range_end).unwrap();
```